crates/ty_python_semantic/src/types/infer.rs

```diff
@@ -8555,14 +8567,35 @@ impl<'db> TypeInferenceBuilder<'db, '_> {
             // and stating that it is only valid as first argument to `typing.Callable[]`
             ast::Expr::List(list) => {
                 self.infer_list_expression(list);
-                self.report_invalid_type_expression(
+                self.report_invalid_type_expression_with_hint(
                     expression,
                     format_args!(
                         "List literals are not allowed in this context in a type expression"
                     ),
+                    Some(format_args!("Did you mean to use `list[...]` instead?")),
```

Comment:
I'd lean towards _only_ adding this hint if the list has 1 item in the slice. If it's `[int]`, it's very likely that the user wanted `list[int]`, but if it's something like `[int, str]`, it feels to me just as likely that the user wanted `tuple[int, str]`.

It would also be great if we could say "Did you mean `list[int]`?" in the suggestion if it was `[int]`, rather than just "Did you mean to use `list[...]`?"!